alchemist-loading/src/main/kotlin/it/unibo/alchemist/loader/export/AbstractExporter.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2010-2021, Danilo Pianini and contributors
+ * listed in the main project's alchemist/build.gradle.kts file.
+ *
+ * This file is part of Alchemist, and is distributed under the terms of the
+ * GNU General Public License, with a linking exception,
+ * as described in the file LICENSE in the Alchemist distribution's top directory.
+ */
+
+package it.unibo.alchemist.loader.export
+
+import it.unibo.alchemist.loader.variables.Variable
+import it.unibo.alchemist.model.interfaces.Environment
+import it.unibo.alchemist.model.interfaces.Position
+import it.unibo.alchemist.model.interfaces.Reaction
+import it.unibo.alchemist.model.interfaces.Time
+import java.util.stream.Collectors
+
+/**
+ * Abstract implementation of a [GenericExporter].
+ * @param samplingInterval the sampling time, defaults to [DEFAULT_INTERVAL].
+ */
+abstract class AbstractExporter<T, P : Position<P>> (
+    private val samplingInterval: Double
+) : GenericExporter<T, P> {
+
+    override var dataExtractor: List<Extractor> = ArrayList()
+
+    /**
+     * The 0th should be sampled.
+     */
+    var count = -1L
```

Comment:
I do not understand what this is and why it is public

Reply:
Done: added documentation.